lib/src/config/config.rs

```diff
@@ -790,6 +794,9 @@ impl ClientConfigBuilder {
         // Configure database
         self.database(config_file.database.clone());
 
+        // Configure the zk prover
+        self.zkp_prover_node_functionality = Some(config_file.zkp_prover_node_functionality);
```

Comment:
I don't think we need an Option here.
This requires changing at a couple of places, though.
```suggestion
        self.zkp_prover_node_functionality = config_file.zkp_prover_node_functionality;
```